File: python/2021/july/lowest-common-ancestor-of-binary-search-tree.py
class Solution:
    def lowestCommonAncestor(self, root, p, q):
        next = (
            p.val < root.val > q.val
            and root.left
            or p.val > root.val < q.val
            and root.right
        )
        return self.lowestCommonAncestor(next, p, q) if next else root

File: go/2021/july/lowest-common-ancestor-of-binary-search-tree.go
package july

import "leetcode/common"

func lowest(root, p, q *common.TreeNode) *common.TreeNode {
	if p == nil || q == nil || root == nil {
		return nil
	}
	if p.Val < root.Val && q.Val < root.Val {
		return lowest(root.Left, p, q)
	}
	if p.Val > root.Val && q.Val > root.Val {
		return lowest(root.Right, p, q)
	}
	return root
}
